### PR TITLE
Untangle: Themes to core & add Theme Showcase submenu

### PR DESCRIPTION
**Pull request by lupus2k**
Fixes https://github.com/Automattic/dotcom-forge/issues/5263, https://github.com/Automattic/dotcom-forge/issues/5264

## Proposed changes:
Remove Calypso links in favor of core links inside `Appearance` menu, and add `Theme Showcase` linked to `wordpress.com/themes/:site_slug`

## Jetpack product discussion
paYJgx-4uN-p2#comment-4524

## Does this pull request change what data or activity we track or use?
NO

## Testing instructions:
* Set your `Admin interface style` to `Classic (wp-admin)` in `wordpress.com/hosting-config/site_slug`
* Go to `wp-admin`
* The `Appareance` sub-menu should link to `/wp-admin` (except 👇 )
* The `Add new link` sub-menu should link to Themes marketplace on Calypso

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php [hunk 1]
```diff
@@ -31,7 +31,11 @@ public function reregister_menu_items() {
 		$this->add_testimonials_menu();
 		$this->add_portfolio_menu();
 		$this->add_comments_menu();
-		$this->add_appearance_menu();
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
```
Comment (candy02058912) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: I'm thinking more of a overriding function in `class-atomic-admin-menu.php` like this (haven't tested):
```
public function add_appearance_menu() {
  if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
     add_submenu_page( 'themes.php', esc_attr__( 'Add new theme', 'jetpack' ), __( 'Add new theme', 'jetpack' ), 'read', 'https://wordpress.com/themes/' . $this->domain );
  } else {
    parent::add_appearance_menu();
  }
}
```

So we can keep the `class-admin-menu.php` cleaner.

**Review by candy02058912 — CHANGES_REQUESTED** [review]
As far as I'm aware, this change should be atomic only + the interface set to classic (or perhaps there's a conversation that I didn't keep up with). So proposing we move the code inside `class-atomic-admin-menu.php`

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php (hunk 1 above)
Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: It works great @candy02058912, thanks for your suggestion! 🎉

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php [hunk 2]
```diff
@@ -30,7 +30,6 @@ public function reregister_menu_items() {
 		$this->add_page_menu();
 		$this->add_testimonials_menu();
 		$this->add_portfolio_menu();
-		$this->add_comments_menu();
```
Comment (DustyReagan) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: Is this deletion intentional? It doesn't seem related to themes.

Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: Definitely not, thanks for noticing 🤦

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 3]
```diff
@@ -523,4 +523,16 @@ public function add_settings_page_notice() {
...
 		add_action( 'admin_notices', $admin_notices );
 	}
+
+	/**
+	 * Adds Appearance menu.
+	 */
+	public function add_appearance_menu() {
+		// When the interface is set to wp-admin, we need to add a link to the Marketplace and rest of the menu keeps like core.
+		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
+			add_submenu_page( 'themes.php', esc_attr__( 'Add new theme', 'jetpack' ), __( 'Add new theme', 'jetpack' ), 'read', 'https://wordpress.com/themes/' . $this->domain );
```
Comment (DustyReagan) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]:
```suggestion
			add_submenu_page( 'themes.php', esc_attr__( 'Add New Theme', 'jetpack' ), __( 'Add New Theme', 'jetpack' ), 'read', 'https://wordpress.com/themes/' . $this->domain );
```
We should capitalize this for consistency with other menu items.

Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php]: Good suggestion, thanks 🎉

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php (hunk 2 above)
Comment (lupus2k) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php]: Updated now 😉

**Review by okmttdhr — DISMISSED** [review]
Teste and worked as explained 👍 (Confirmed all the comments are addressed)

**Comment by okmttdhr** [comment]
Is it appropriate to change the link of `Add New Theme` to `https://wordpress.com/themes/<site-slug>` on the `/wp-admin/themes.php` page, which currently redirects users to `/wp-admin/theme-install.php`?
Screenshot 2024-01-23 at 16 26 58

Similarly, should we keep the link of `Install new theme` on the `https://wordpress.com/themes/<site-slug>`, which currently redirects users to `/wp-admin/theme-install.php`?
Screenshot 2024-01-23 at 16 25 30

Please correct me if I am mistaken, as I might be missing some context. 🙏 😃

**Comment by lupus2k** [comment]
Hey @okmttdhr, we started a discussion about it here: p1705964787729999-slack-C06DN6QQVAQ

**Review by DustyReagan — DISMISSED** [review]
Looks good! Just need to rename the new submenu item to “Theme Showcase”
p1706026646168939/1705964787.729999-slack-C06DN6QQVAQ

**Comment by lupus2k** [comment]
Renamed `Add New Theme` to `Theme Showcase` and link it to Calypso Themes

**Review by DustyReagan — APPROVED** [review]

**Review by candy02058912 — APPROVED** [review]
LGTM, thanks for making the changes 👍
